app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-search-tracker.js

```diff
@@ -35,16 +42,24 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
       this.$module.addEventListener('submit', event => this.trackSearch(event))
     }
 
+    setTriggeredAction (event) {
+      if (event.target.type === 'search') {
+        this.triggeredAction = 'search'
+      } else if (this.triggeredAction !== 'search') {
+        // The 'search' action always takes precedence over the 'filter' action, so only set the
+        // action to 'filter' if it is not already 'search'.
+        this.triggeredAction = 'filter'
+      }
+    }
```

Comment:
Does there need to be some code to set the `triggeredAction` back to `filter` if the user did modify the search input, but then undoes the change, so in the end they didn't actually change the keyword?

Reply:
Good question!

We ended up deciding for the sake of simplicity to treat any change of search or filters as a done deal, even if the user then changes the fields back to their original value (so we're measuring _intent_ rather than what they end up doing). We may decide to be more specific about it in the future though.